{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Feature engineering: Aggregating sparse variant data\n",
    "\n",
    "In this version of feature engineering, we follow the steps of the paper.\n",
    "\n",
    "\n",
    "## Starting Dataset:\n",
    "- Variant calls from 466 genes (SNVs, indel) and fusions in 3 genes.\n",
    "  - Highly sparse data with few shared exact variants across samples.\n",
    "\n",
    "\n",
    "## Feature Aggregation Approach:\n",
    "\n",
    "- Data transformed to gene-level features\n",
    "   - SNVs, indels and fusions (SV) handled separately.\n",
    "   - Allele frequency (AF) values ignored\n",
    "   - Binary representation indicating presence or absence of alteration.\n",
    "   - example features: EZH2_SNV, PIM1_indel, BCL2_SV\n",
    "   - MYD88 L273P mutation was singled out as a key driver for ABC DLBCL\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# python module loading\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import datetime"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "## SNV data\n",
    "\n",
    "\n",
    "From the [supplemantatry](https://ehsantabari.com/public/goya_paper/GOYA_ctDNA_supplementary_methods_Leuk_Lymph_31Oct23.pdf) methods:\n",
    "\n",
    "> ### Variant Filtering\n",
    ">\n",
    "> Common variants in the Single Nucleotide Polymorphism Database (dbSNP) or variants with an\n",
    "> AF > 0.1% in any of 1000 Genomes project (1KG) or Exome Aggregation Consortium (ExaC)\n",
    "> populations were filtered out unless they were reported by the Catalogue of Somatic Mutations in\n",
    "> Cancer (COSMIC) or the Cancer Genome Atlas Program (TCGA). Variants that were detected in\n",
    "> low complexity regions of the genome, as well as regions known to contain immunoglobulin\n",
    "> genes, were filtered out. Variants that were commonly detected using our targeted panel on\n",
    "> healthy normal plasma samples were also discarded. For this blacklist filter, 22 healthy donor\n",
    "> samples were sequenced with our workflow, and any variants occurring in more than two healthy\n",
    "> donors with more than five supporting reads were excluded from consideration in patients with\n",
    "> DLBCL. Finally, variants with low coverage (a depth of < 25% of median sample depth) were\n",
    "> filtered out. After filtering, 58961 SNVs and 2954 indels were detected in plasma samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Sample number', 'CHR', 'POS', 'REF', 'ALT', 'Gene', 'CDS', 'AA',\n",
      "       'NONSILENT', 'TOTDEPTH', 'VARDEPTH'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample number</th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CDS</th>\n",
       "      <th>AA</th>\n",
       "      <th>NONSILENT</th>\n",
       "      <th>TOTDEPTH</th>\n",
       "      <th>VARDEPTH</th>\n",
       "      <th>feature_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2558453</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>c.289C&gt;A</td>\n",
       "      <td>p.Gln97Lys</td>\n",
       "      <td>True</td>\n",
       "      <td>7672</td>\n",
       "      <td>5</td>\n",
       "      <td>TNFRSF14_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>23559151</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ID3</td>\n",
       "      <td>c.276T&gt;C</td>\n",
       "      <td>p.Pro92Pro</td>\n",
       "      <td>False</td>\n",
       "      <td>7234</td>\n",
       "      <td>5</td>\n",
       "      <td>ID3_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>85270733</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>BCL10</td>\n",
       "      <td>c.231A&gt;G</td>\n",
       "      <td>p.Lys77Lys</td>\n",
       "      <td>False</td>\n",
       "      <td>7009</td>\n",
       "      <td>7</td>\n",
       "      <td>BCL10_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>474411</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NTNG1</td>\n",
       "      <td>c.376T&gt;A</td>\n",
       "      <td>p.Tyr126Asn</td>\n",
       "      <td>True</td>\n",
       "      <td>7408</td>\n",
       "      <td>4</td>\n",
       "      <td>NTNG1_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>203306270</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>BTG2</td>\n",
       "      <td>c.142+522T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>6442</td>\n",
       "      <td>17</td>\n",
       "      <td>BTG2_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58956</th>\n",
       "      <td>310</td>\n",
       "      <td>chr9</td>\n",
       "      <td>5534930</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PDCD1LG2</td>\n",
       "      <td>c.241C&gt;T</td>\n",
       "      <td>p.Pro81Ser</td>\n",
       "      <td>True</td>\n",
       "      <td>1889</td>\n",
       "      <td>899</td>\n",
       "      <td>PDCD1LG2_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58957</th>\n",
       "      <td>310</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89332613</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>IFIT3</td>\n",
       "      <td>c.-10A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1492</td>\n",
       "      <td>753</td>\n",
       "      <td>IFIT3_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58958</th>\n",
       "      <td>310</td>\n",
       "      <td>chr14</td>\n",
       "      <td>105645665</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>RP11-731F5.2</td>\n",
       "      <td>n.105645665G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>1670</td>\n",
       "      <td>365</td>\n",
       "      <td>RP11-731F5.2_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58959</th>\n",
       "      <td>310</td>\n",
       "      <td>chr14</td>\n",
       "      <td>105774618</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>IGHG1</td>\n",
       "      <td>c.437-31548G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2666</td>\n",
       "      <td>93</td>\n",
       "      <td>IGHG1_SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58960</th>\n",
       "      <td>310</td>\n",
       "      <td>chr14</td>\n",
       "      <td>106627850</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>IGHV4-59</td>\n",
       "      <td>c.-170A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2744</td>\n",
       "      <td>964</td>\n",
       "      <td>IGHV4-59_SNV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58961 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample number    CHR        POS REF ALT          Gene             CDS  \\\n",
       "0                  1   chr1    2558453   C   A      TNFRSF14        c.289C>A   \n",
       "1                  1   chr1   23559151   A   G           ID3        c.276T>C   \n",
       "2                  1   chr1   85270733   T   C         BCL10        c.231A>G   \n",
       "3                  1   chr1     474411   T   A         NTNG1        c.376T>A   \n",
       "4                  1   chr1  203306270   T   G          BTG2    c.142+522T>G   \n",
       "...              ...    ...        ...  ..  ..           ...             ...   \n",
       "58956            310   chr9    5534930   C   T      PDCD1LG2        c.241C>T   \n",
       "58957            310  chr10   89332613   A   G         IFIT3        c.-10A>G   \n",
       "58958            310  chr14  105645665   C   A  RP11-731F5.2  n.105645665G>T   \n",
       "58959            310  chr14  105774618   C   T         IGHG1  c.437-31548G>A   \n",
       "58960            310  chr14  106627850   T   C      IGHV4-59       c.-170A>G   \n",
       "\n",
       "                AA  NONSILENT  TOTDEPTH  VARDEPTH      feature_name  \n",
       "0       p.Gln97Lys       True      7672         5      TNFRSF14_SNV  \n",
       "1       p.Pro92Pro      False      7234         5           ID3_SNV  \n",
       "2       p.Lys77Lys      False      7009         7         BCL10_SNV  \n",
       "3      p.Tyr126Asn       True      7408         4         NTNG1_SNV  \n",
       "4              NaN      False      6442        17          BTG2_SNV  \n",
       "...            ...        ...       ...       ...               ...  \n",
       "58956   p.Pro81Ser       True      1889       899      PDCD1LG2_SNV  \n",
       "58957          NaN      False      1492       753         IFIT3_SNV  \n",
       "58958          NaN      False      1670       365  RP11-731F5.2_SNV  \n",
       "58959          NaN      False      2666        93         IGHG1_SNV  \n",
       "58960          NaN      False      2744       964      IGHV4-59_SNV  \n",
       "\n",
       "[58961 rows x 12 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Loading SNV data\n",
    "snv = pd.read_excel(\n",
    "    \"https://ehsantabari.com/public/goya_paper/Supplementary_Table_2A.xlsx\"\n",
    ")\n",
    "\n",
    "\n",
    "print(snv.columns)\n",
    "\n",
    "\n",
    "# Methods:\n",
    "# \"In one case, a variant within the MYD88 gene (L265P/L273P) was singled out as it is a known common driver for ABC DLBCL\"\n",
    "snv[\"feature_name\"] = snv[\"Gene\"].astype(str)\n",
    "snv.loc[(snv[\"Gene\"] == \"MYD88\") & (snv[\"AA\"] == \"p.Leu273Pro\"), \"feature_name\"] = (\n",
    "    \"MYD88_L273P\"\n",
    ")\n",
    "\n",
    "# chr17: 77451809-77452328 is misannotated; Fix Gene name\n",
    "snv.loc[\n",
    "    (snv[\"CHR\"] == \"chr17\") & (snv[\"POS\"] >= 77451809) & (snv[\"POS\"] <= 77452328),\n",
    "    \"feature_name\",\n",
    "] = \"SEPT9\"\n",
    "\n",
    "snv[\"feature_name\"] = snv[\"feature_name\"] + \"_SNV\"\n",
    "\n",
    "snv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The number of variants are post filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>feature_name</th>\n",
       "      <th>Sample number</th>\n",
       "      <th>ABCB11_SNV</th>\n",
       "      <th>AC004623.2_SNV</th>\n",
       "      <th>AC007250.4_SNV</th>\n",
       "      <th>AC007386.4_SNV</th>\n",
       "      <th>AC0142.1_SNV</th>\n",
       "      <th>AC092170.1-AC009312.1_SNV</th>\n",
       "      <th>AC096579.13_SNV</th>\n",
       "      <th>AC118562.1-RP11-427M20.1_SNV</th>\n",
       "      <th>AC245028.1_SNV</th>\n",
       "      <th>...</th>\n",
       "      <th>ZEB2_SNV</th>\n",
       "      <th>ZFP36L1_SNV</th>\n",
       "      <th>ZFP42_SNV</th>\n",
       "      <th>ZMYM6_SNV</th>\n",
       "      <th>ZNF160_SNV</th>\n",
       "      <th>ZNF577_SNV</th>\n",
       "      <th>ZNF578_SNV</th>\n",
       "      <th>ZNF608_SNV</th>\n",
       "      <th>ZNF649_SNV</th>\n",
       "      <th>ZNF678_SNV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>306</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>307</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>308</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>310</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>310 rows Ã— 480 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "feature_name  Sample number  ABCB11_SNV  AC004623.2_SNV  AC007250.4_SNV  \\\n",
       "0                         1         0.0             0.0             0.0   \n",
       "1                         2         0.0             0.0             0.0   \n",
       "2                         3         0.0             0.0             0.0   \n",
       "3                         4         0.0             0.0             0.0   \n",
       "4                         5         0.0             0.0             0.0   \n",
       "..                      ...         ...             ...             ...   \n",
       "305                     306         0.0             0.0             0.0   \n",
       "306                     307         0.0             0.0             0.0   \n",
       "307                     308         0.0             0.0             0.0   \n",
       "308                     309         0.0             0.0             0.0   \n",
       "309                     310         0.0             0.0             0.0   \n",
       "\n",
       "feature_name  AC007386.4_SNV  AC0142.1_SNV  AC092170.1-AC009312.1_SNV  \\\n",
       "0                        0.0           0.0                        0.0   \n",
       "1                        0.0           0.0                        0.0   \n",
       "2                        0.0           0.0                        0.0   \n",
       "3                        0.0           0.0                        0.0   \n",
       "4                        0.0           0.0                        0.0   \n",
       "..                       ...           ...                        ...   \n",
       "305                      0.0           0.0                        0.0   \n",
       "306                      0.0           0.0                        0.0   \n",
       "307                      0.0           0.0                        0.0   \n",
       "308                      0.0           0.0                        0.0   \n",
       "309                      0.0           0.0                        0.0   \n",
       "\n",
       "feature_name  AC096579.13_SNV  AC118562.1-RP11-427M20.1_SNV  AC245028.1_SNV  \\\n",
       "0                         0.0                           0.0             0.0   \n",
       "1                         0.0                           0.0             0.0   \n",
       "2                         1.0                           0.0             0.0   \n",
       "3                         1.0                           0.0             0.0   \n",
       "4                         1.0                           0.0             0.0   \n",
       "..                        ...                           ...             ...   \n",
       "305                       1.0                           0.0             0.0   \n",
       "306                       1.0                           0.0             0.0   \n",
       "307                       1.0                           0.0             0.0   \n",
       "308                       1.0                           0.0             0.0   \n",
       "309                       0.0                           0.0             0.0   \n",
       "\n",
       "feature_name  ...  ZEB2_SNV  ZFP36L1_SNV  ZFP42_SNV  ZMYM6_SNV  ZNF160_SNV  \\\n",
       "0             ...       0.0          1.0        0.0        0.0         0.0   \n",
       "1             ...       0.0          1.0        0.0        0.0         0.0   \n",
       "2             ...       1.0          0.0        0.0        0.0         1.0   \n",
       "3             ...       0.0          1.0        1.0        0.0         0.0   \n",
       "4             ...       0.0          1.0        0.0        0.0         0.0   \n",
       "..            ...       ...          ...        ...        ...         ...   \n",
       "305           ...       0.0          1.0        0.0        0.0         0.0   \n",
       "306           ...       0.0          0.0        0.0        0.0         0.0   \n",
       "307           ...       0.0          0.0        0.0        0.0         0.0   \n",
       "308           ...       0.0          1.0        0.0        0.0         0.0   \n",
       "309           ...       0.0          0.0        0.0        0.0         0.0   \n",
       "\n",
       "feature_name  ZNF577_SNV  ZNF578_SNV  ZNF608_SNV  ZNF649_SNV  ZNF678_SNV  \n",
       "0                    0.0         0.0         0.0         0.0         0.0  \n",
       "1                    0.0         0.0         0.0         0.0         0.0  \n",
       "2                    1.0         0.0         1.0         0.0         0.0  \n",
       "3                    0.0         0.0         1.0         0.0         0.0  \n",
       "4                    0.0         0.0         0.0         0.0         0.0  \n",
       "..                   ...         ...         ...         ...         ...  \n",
       "305                  0.0         0.0         0.0         0.0         0.0  \n",
       "306                  0.0         0.0         0.0         0.0         0.0  \n",
       "307                  0.0         0.0         0.0         0.0         1.0  \n",
       "308                  0.0         1.0         0.0         0.0         0.0  \n",
       "309                  0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[310 rows x 480 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For each sample, a list of genes with any non-silent somatic SNVs, indels or translocations was compiled,\n",
    "# regardless of the loci of the variants within the gene; for example, SNV in EZH2, indel in PIM1 or translocation in BCL2.\n",
    "\n",
    "snv_data = snv.groupby([\"Sample number\", \"feature_name\"]).size().reset_index()\n",
    "snv_data[\"value\"] = 1\n",
    "snv_data = snv_data.pivot(\n",
    "    index=\"Sample number\", columns=\"feature_name\", values=\"value\"\n",
    ").fillna(0)\n",
    "snv_data = snv_data.reset_index()\n",
    "snv_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Indel data\n",
    "\n",
    "We are expecting to find that \n",
    "> ... _2954 indels were detected in plasma samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Sample number ', 'CHR', 'POS', 'REF', 'ALT', 'Gene', 'CDS', 'AA',\n",
      "       'TOTDEPTH', 'VARDEPTH'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample number</th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>CDS</th>\n",
       "      <th>AA</th>\n",
       "      <th>TOTDEPTH</th>\n",
       "      <th>VARDEPTH</th>\n",
       "      <th>feature_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>96144233</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>DUSP2</td>\n",
       "      <td>c.650delT</td>\n",
       "      <td>p.Phe217fs</td>\n",
       "      <td>9341</td>\n",
       "      <td>19</td>\n",
       "      <td>DUSP2_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>chr8</td>\n",
       "      <td>127738385</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>MYC</td>\n",
       "      <td>c.173delC</td>\n",
       "      <td>p.Pro58fs</td>\n",
       "      <td>7114</td>\n",
       "      <td>15</td>\n",
       "      <td>MYC_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>49022330</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>c.404delG</td>\n",
       "      <td>p.Arg135fs</td>\n",
       "      <td>6071</td>\n",
       "      <td>13</td>\n",
       "      <td>KMT2D_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>chr16</td>\n",
       "      <td>3729196</td>\n",
       "      <td>GCT</td>\n",
       "      <td>G</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>c.5849_5850delAG</td>\n",
       "      <td>p.Gln1950fs</td>\n",
       "      <td>3073</td>\n",
       "      <td>12</td>\n",
       "      <td>CREBBP_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>65056349</td>\n",
       "      <td>TCG</td>\n",
       "      <td>T</td>\n",
       "      <td>G13</td>\n",
       "      <td>c.243_244delCG</td>\n",
       "      <td>p.Glu82fs</td>\n",
       "      <td>2864</td>\n",
       "      <td>7</td>\n",
       "      <td>G13_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949</th>\n",
       "      <td>309</td>\n",
       "      <td>chrX</td>\n",
       "      <td>74742737</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>NEXMIF</td>\n",
       "      <td>c.1819delT</td>\n",
       "      <td>p.Ser607fs</td>\n",
       "      <td>4214</td>\n",
       "      <td>9</td>\n",
       "      <td>NEXMIF_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950</th>\n",
       "      <td>309</td>\n",
       "      <td>chrX</td>\n",
       "      <td>87518322</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>KLHL4</td>\n",
       "      <td>c.422+8delT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2434</td>\n",
       "      <td>12</td>\n",
       "      <td>KLHL4_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2951</th>\n",
       "      <td>310</td>\n",
       "      <td>chr6</td>\n",
       "      <td>137871203</td>\n",
       "      <td>TCC</td>\n",
       "      <td>T</td>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>c.-15-9_-15-8delCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>739</td>\n",
       "      <td>6</td>\n",
       "      <td>TNFAIP3_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2952</th>\n",
       "      <td>310</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7673747</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>c.872delA</td>\n",
       "      <td>p.Lys291fs</td>\n",
       "      <td>5104</td>\n",
       "      <td>12</td>\n",
       "      <td>TP53_INDEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2953</th>\n",
       "      <td>310</td>\n",
       "      <td>chrX</td>\n",
       "      <td>142203778</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>c.209delA</td>\n",
       "      <td>p.Glu70fs</td>\n",
       "      <td>624</td>\n",
       "      <td>6</td>\n",
       "      <td>MAGEC2_INDEL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2954 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample number    CHR        POS  REF ALT     Gene                 CDS  \\\n",
       "0                 1   chr2   96144233   GA   G    DUSP2           c.650delT   \n",
       "1                 1   chr8  127738385   GC   G      MYC           c.173delC   \n",
       "2                 1  chr12   49022330   TC   T    KMT2D           c.404delG   \n",
       "3                 1  chr16    3729196  GCT   G   CREBBP    c.5849_5850delAG   \n",
       "4                 1  chr17   65056349  TCG   T      G13      c.243_244delCG   \n",
       "...             ...    ...        ...  ...  ..      ...                 ...   \n",
       "2949            309   chrX   74742737   GA   G   NEXMIF          c.1819delT   \n",
       "2950            309   chrX   87518322   GT   G    KLHL4         c.422+8delT   \n",
       "2951            310   chr6  137871203  TCC   T  TNFAIP3  c.-15-9_-15-8delCC   \n",
       "2952            310  chr17    7673747   CT   C     TP53           c.872delA   \n",
       "2953            310   chrX  142203778   CT   C   MAGEC2           c.209delA   \n",
       "\n",
       "               AA  TOTDEPTH  VARDEPTH   feature_name  \n",
       "0      p.Phe217fs      9341        19    DUSP2_INDEL  \n",
       "1       p.Pro58fs      7114        15      MYC_INDEL  \n",
       "2      p.Arg135fs      6071        13    KMT2D_INDEL  \n",
       "3     p.Gln1950fs      3073        12   CREBBP_INDEL  \n",
       "4       p.Glu82fs      2864         7      G13_INDEL  \n",
       "...           ...       ...       ...            ...  \n",
       "2949   p.Ser607fs      4214         9   NEXMIF_INDEL  \n",
       "2950          NaN      2434        12    KLHL4_INDEL  \n",
       "2951          NaN       739         6  TNFAIP3_INDEL  \n",
       "2952   p.Lys291fs      5104        12     TP53_INDEL  \n",
       "2953    p.Glu70fs       624         6   MAGEC2_INDEL  \n",
       "\n",
       "[2954 rows x 11 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Loading SNV data\n",
    "indel = pd.read_excel(\n",
    "    \"https://ehsantabari.com/public/goya_paper/Supplementary_Table_2B.xlsx\"\n",
    ")\n",
    "\n",
    "print(indel.columns)\n",
    "\n",
    "# There is a space after the sample number column name: Fix it\n",
    "indel.rename(columns={\"Sample number \": \"Sample number\"}, inplace=True)\n",
    "\n",
    "indel[\"feature_name\"] = indel[\"Gene\"] + \"_INDEL\"\n",
    "\n",
    "indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>feature_name</th>\n",
       "      <th>Sample number</th>\n",
       "      <th>ABCB11_INDEL</th>\n",
       "      <th>AC072022.1_INDEL</th>\n",
       "      <th>ACAD8_INDEL</th>\n",
       "      <th>ACTA2_INDEL</th>\n",
       "      <th>ACTB_INDEL</th>\n",
       "      <th>ACTG1_INDEL</th>\n",
       "      <th>ACVR2A_INDEL</th>\n",
       "      <th>ADAMTS1_INDEL</th>\n",
       "      <th>AL928742.1_INDEL</th>\n",
       "      <th>...</th>\n",
       "      <th>TNFAIP3_INDEL</th>\n",
       "      <th>TNFRSF14_INDEL</th>\n",
       "      <th>TP53_INDEL</th>\n",
       "      <th>TSPOAP1-AS1_INDEL</th>\n",
       "      <th>XBP1_INDEL</th>\n",
       "      <th>ZEB2_INDEL</th>\n",
       "      <th>ZFP36L1_INDEL</th>\n",
       "      <th>ZFP42_INDEL</th>\n",
       "      <th>ZNF608_INDEL</th>\n",
       "      <th>ZNF649_INDEL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>306</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>308</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>310</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>291 rows Ã— 193 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "feature_name  Sample number  ABCB11_INDEL  AC072022.1_INDEL  ACAD8_INDEL  \\\n",
       "0                         1           0.0               0.0          0.0   \n",
       "1                         2           0.0               0.0          0.0   \n",
       "2                         3           0.0               0.0          0.0   \n",
       "3                         4           0.0               0.0          0.0   \n",
       "4                         5           0.0               0.0          0.0   \n",
       "..                      ...           ...               ...          ...   \n",
       "286                     305           0.0               0.0          0.0   \n",
       "287                     306           0.0               0.0          0.0   \n",
       "288                     308           0.0               0.0          0.0   \n",
       "289                     309           0.0               0.0          0.0   \n",
       "290                     310           0.0               0.0          0.0   \n",
       "\n",
       "feature_name  ACTA2_INDEL  ACTB_INDEL  ACTG1_INDEL  ACVR2A_INDEL  \\\n",
       "0                     0.0         0.0          0.0           0.0   \n",
       "1                     0.0         0.0          0.0           0.0   \n",
       "2                     1.0         0.0          0.0           0.0   \n",
       "3                     0.0         0.0          0.0           0.0   \n",
       "4                     0.0         1.0          0.0           0.0   \n",
       "..                    ...         ...          ...           ...   \n",
       "286                   0.0         0.0          0.0           0.0   \n",
       "287                   0.0         0.0          0.0           0.0   \n",
       "288                   0.0         0.0          0.0           0.0   \n",
       "289                   0.0         0.0          0.0           0.0   \n",
       "290                   0.0         0.0          0.0           0.0   \n",
       "\n",
       "feature_name  ADAMTS1_INDEL  AL928742.1_INDEL  ...  TNFAIP3_INDEL  \\\n",
       "0                       0.0               0.0  ...            0.0   \n",
       "1                       0.0               0.0  ...            1.0   \n",
       "2                       0.0               0.0  ...            0.0   \n",
       "3                       0.0               0.0  ...            0.0   \n",
       "4                       0.0               0.0  ...            0.0   \n",
       "..                      ...               ...  ...            ...   \n",
       "286                     0.0               0.0  ...            0.0   \n",
       "287                     0.0               0.0  ...            0.0   \n",
       "288                     0.0               0.0  ...            0.0   \n",
       "289                     0.0               0.0  ...            1.0   \n",
       "290                     0.0               0.0  ...            1.0   \n",
       "\n",
       "feature_name  TNFRSF14_INDEL  TP53_INDEL  TSPOAP1-AS1_INDEL  XBP1_INDEL  \\\n",
       "0                        0.0         0.0                0.0         0.0   \n",
       "1                        0.0         0.0                0.0         0.0   \n",
       "2                        0.0         0.0                0.0         0.0   \n",
       "3                        0.0         0.0                1.0         0.0   \n",
       "4                        0.0         0.0                0.0         0.0   \n",
       "..                       ...         ...                ...         ...   \n",
       "286                      0.0         0.0                0.0         0.0   \n",
       "287                      0.0         0.0                0.0         0.0   \n",
       "288                      0.0         0.0                0.0         0.0   \n",
       "289                      0.0         0.0                0.0         0.0   \n",
       "290                      0.0         1.0                0.0         0.0   \n",
       "\n",
       "feature_name  ZEB2_INDEL  ZFP36L1_INDEL  ZFP42_INDEL  ZNF608_INDEL  \\\n",
       "0                    0.0            0.0          0.0           0.0   \n",
       "1                    0.0            0.0          0.0           0.0   \n",
       "2                    0.0            0.0          0.0           0.0   \n",
       "3                    0.0            0.0          0.0           0.0   \n",
       "4                    0.0            0.0          0.0           0.0   \n",
       "..                   ...            ...          ...           ...   \n",
       "286                  0.0            0.0          0.0           0.0   \n",
       "287                  0.0            0.0          0.0           0.0   \n",
       "288                  0.0            0.0          0.0           0.0   \n",
       "289                  0.0            0.0          0.0           0.0   \n",
       "290                  0.0            0.0          0.0           0.0   \n",
       "\n",
       "feature_name  ZNF649_INDEL  \n",
       "0                      0.0  \n",
       "1                      0.0  \n",
       "2                      0.0  \n",
       "3                      0.0  \n",
       "4                      0.0  \n",
       "..                     ...  \n",
       "286                    0.0  \n",
       "287                    0.0  \n",
       "288                    0.0  \n",
       "289                    0.0  \n",
       "290                    0.0  \n",
       "\n",
       "[291 rows x 193 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For each sample, a list of genes with any non-silent somatic SNVs, indels or translocations was compiled,\n",
    "# regardless of the loci of the variants within the gene; for example, SNV in EZH2, indel in PIM1 or translocation in BCL2.\n",
    "\n",
    "indel_data = indel.groupby([\"Sample number\", \"feature_name\"]).size().reset_index()\n",
    "indel_data[\"value\"] = 1\n",
    "indel_data = indel_data.pivot(\n",
    "    index=\"Sample number\", columns=\"feature_name\", values=\"value\"\n",
    ").fillna(0)\n",
    "indel_data = indel_data.reset_index()\n",
    "indel_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fusion data\n",
    "\n",
    "We are expecting to find that \n",
    "\n",
    "> In addition, 170 translocations of BCL2, BCL6 and MYC genes were detected (Table S2)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Sample number', 'Gene', 'Gene (other)', 'Est_Type', 'Total depth',\n",
      "       'Break depth', 'Break support1', 'Break support2',\n",
      "       'Proper pair support'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample number</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene (other)</th>\n",
       "      <th>Est_Type</th>\n",
       "      <th>Total depth</th>\n",
       "      <th>Break depth</th>\n",
       "      <th>Break support1</th>\n",
       "      <th>Break support2</th>\n",
       "      <th>Proper pair support</th>\n",
       "      <th>feature_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>IGHG2</td>\n",
       "      <td>TRA</td>\n",
       "      <td>46462</td>\n",
       "      <td>33</td>\n",
       "      <td>21</td>\n",
       "      <td>30</td>\n",
       "      <td>31</td>\n",
       "      <td>BCL6_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>IRF8</td>\n",
       "      <td>TRA</td>\n",
       "      <td>23682</td>\n",
       "      <td>3315</td>\n",
       "      <td>2731</td>\n",
       "      <td>1</td>\n",
       "      <td>3308</td>\n",
       "      <td>BCL6_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>IRF1</td>\n",
       "      <td>TRA</td>\n",
       "      <td>19445</td>\n",
       "      <td>90</td>\n",
       "      <td>54</td>\n",
       "      <td>41</td>\n",
       "      <td>90</td>\n",
       "      <td>BCL6_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>IGHJ6</td>\n",
       "      <td>TRA</td>\n",
       "      <td>27840</td>\n",
       "      <td>288</td>\n",
       "      <td>2</td>\n",
       "      <td>636</td>\n",
       "      <td>270</td>\n",
       "      <td>BCL2_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>IGLJ2</td>\n",
       "      <td>TRA</td>\n",
       "      <td>5201</td>\n",
       "      <td>657</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>648</td>\n",
       "      <td>BCL2_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>303</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>MIR4537</td>\n",
       "      <td>TRA</td>\n",
       "      <td>18066</td>\n",
       "      <td>1213</td>\n",
       "      <td>988</td>\n",
       "      <td>2</td>\n",
       "      <td>1190</td>\n",
       "      <td>BCL6_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>303</td>\n",
       "      <td>MYC</td>\n",
       "      <td>IGHG4</td>\n",
       "      <td>TRA</td>\n",
       "      <td>21952</td>\n",
       "      <td>743</td>\n",
       "      <td>467</td>\n",
       "      <td>3</td>\n",
       "      <td>737</td>\n",
       "      <td>MYC_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>306</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>MIR4537</td>\n",
       "      <td>TRA</td>\n",
       "      <td>15650</td>\n",
       "      <td>2432</td>\n",
       "      <td>2015</td>\n",
       "      <td>1542</td>\n",
       "      <td>2410</td>\n",
       "      <td>BCL2_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>308</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>PHLPP1</td>\n",
       "      <td>DEL</td>\n",
       "      <td>23209</td>\n",
       "      <td>1690</td>\n",
       "      <td>1</td>\n",
       "      <td>1413</td>\n",
       "      <td>1688</td>\n",
       "      <td>BCL2_SV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>309</td>\n",
       "      <td>MYC</td>\n",
       "      <td>PCDH19</td>\n",
       "      <td>TRA</td>\n",
       "      <td>16950</td>\n",
       "      <td>140</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>140</td>\n",
       "      <td>MYC_SV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Sample number  Gene Gene (other) Est_Type  Total depth  Break depth  \\\n",
       "0                1  BCL6        IGHG2      TRA        46462           33   \n",
       "1                2  BCL6         IRF8      TRA        23682         3315   \n",
       "2                5  BCL6         IRF1      TRA        19445           90   \n",
       "3                6  BCL2        IGHJ6      TRA        27840          288   \n",
       "4                7  BCL2        IGLJ2      TRA         5201          657   \n",
       "..             ...   ...          ...      ...          ...          ...   \n",
       "165            303  BCL6      MIR4537      TRA        18066         1213   \n",
       "166            303   MYC        IGHG4      TRA        21952          743   \n",
       "167            306  BCL2      MIR4537      TRA        15650         2432   \n",
       "168            308  BCL2       PHLPP1      DEL        23209         1690   \n",
       "169            309   MYC       PCDH19      TRA        16950          140   \n",
       "\n",
       "     Break support1  Break support2  Proper pair support feature_name  \n",
       "0                21              30                   31      BCL6_SV  \n",
       "1              2731               1                 3308      BCL6_SV  \n",
       "2                54              41                   90      BCL6_SV  \n",
       "3                 2             636                  270      BCL2_SV  \n",
       "4                26               4                  648      BCL2_SV  \n",
       "..              ...             ...                  ...          ...  \n",
       "165             988               2                 1190      BCL6_SV  \n",
       "166             467               3                  737       MYC_SV  \n",
       "167            2015            1542                 2410      BCL2_SV  \n",
       "168               1            1413                 1688      BCL2_SV  \n",
       "169              62               1                  140       MYC_SV  \n",
       "\n",
       "[170 rows x 10 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Loading SNV data\n",
    "fusion = pd.read_excel(\n",
    "    \"https://ehsantabari.com/public/goya_paper/Supplementary_Table_2C.xlsx\"\n",
    ")\n",
    "print(fusion.columns)\n",
    "\n",
    "fusion[\"feature_name\"] = fusion[\"Gene\"] + \"_SV\"\n",
    "\n",
    "fusion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>feature_name</th>\n",
       "      <th>Sample number</th>\n",
       "      <th>BCL2_SV</th>\n",
       "      <th>BCL6_SV</th>\n",
       "      <th>MYC_SV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>298</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>303</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>306</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>308</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>119 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "feature_name  Sample number  BCL2_SV  BCL6_SV  MYC_SV\n",
       "0                         1      0.0      1.0     0.0\n",
       "1                         2      0.0      1.0     0.0\n",
       "2                         5      0.0      1.0     0.0\n",
       "3                         6      1.0      0.0     0.0\n",
       "4                         7      1.0      0.0     0.0\n",
       "..                      ...      ...      ...     ...\n",
       "114                     298      0.0      1.0     0.0\n",
       "115                     303      0.0      1.0     1.0\n",
       "116                     306      1.0      0.0     0.0\n",
       "117                     308      1.0      0.0     0.0\n",
       "118                     309      0.0      0.0     1.0\n",
       "\n",
       "[119 rows x 4 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For each sample, a list of genes with any non-silent somatic SNVs, indels or translocations was compiled,\n",
    "# regardless of the loci of the variants within the gene; for example, SNV in EZH2, indel in PIM1 or translocation in BCL2.\n",
    "\n",
    "fusion_data = fusion.groupby([\"Sample number\", \"feature_name\"]).size().reset_index()\n",
    "fusion_data[\"value\"] = 1\n",
    "fusion_data = fusion_data.pivot(\n",
    "    index=\"Sample number\", columns=\"feature_name\", values=\"value\"\n",
    ").fillna(0)\n",
    "fusion_data = fusion_data.reset_index()\n",
    "fusion_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge the dataset and create a data file\n",
    "\n",
    "The merged data will be stored locally for later use as `goya_data-gene_noAf-(datetime).csv`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(310, 675)"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge the three snv, indel and fusion data and save it to a file goya_data-gene_noAf-(datetime).csv \n",
    "goya_data = pd.merge(snv_data, indel_data, on=\"Sample number\", how=\"outer\")\n",
    "goya_data = pd.merge(goya_data, fusion_data, on=\"Sample number\", how=\"outer\")\n",
    "goya_data = goya_data.fillna(0)\n",
    "goya_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add date time to the file name to avoid overwriting\n",
    "# use a file name frirendly format, no need to include the seconds in the file name\n",
    "# now = datetime.datetime.now()\n",
    "now = datetime.datetime.now().strftime(\"%y_%m_%d_%H_%M\")  \n",
    "goya_data.to_csv(f\"goya_data-gene_noAf-{now}.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ml-env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
